ttnn/ttnn/device.py

```diff
@@ -138,11 +138,47 @@ def is_grayskull(device):
     return device.arch() == ttnn._ttnn.device.Arch.GRAYSKULL
 
 
+def is_blackhole(device):
+    return device.arch() == ttnn._ttnn.device.Arch.BLACKHOLE
+
+
 SetDefaultDevice = ttnn._ttnn.device.SetDefaultDevice
 GetDefaultDevice = ttnn._ttnn.device.GetDefaultDevice
 format_input_tensor = ttnn._ttnn.device.format_input_tensor
 format_output_tensor = ttnn._ttnn.device.format_output_tensor
 pad_to_tile_shape = ttnn._ttnn.device.pad_to_tile_shape
 
 
+def GetComputeKernelConfig(
+    math_fidelity=ttnn.MathFidelity.HiFi4,
+    math_approx_mode=True,
+    fp32_dest_acc_en=False,
+    packer_l1_acc=False,
+    dst_full_sync_en=False,
+):
+    device = ttnn.GetDefaultDevice()
```

Comment:
It is not guaranteed that a Default device is set. 
`GetComputeKernelConfig` seems to be used everywhere, making so much code rely on something that is not guaranteed.

Reply:
should this become a device's function?

should it accept device as a parameter? if not provided, function can try to query the default device and if it is not set - throw.